test/parallel/test-worker-init-failure.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const child_process = require('child_process');
+
+// Test that workers fail with meaningful error message
+// when their initialization fails.
+
+if (process.argv[2] === 'child') {
+  const { Worker } = require('worker_threads');
+  for (let i = 0; i < 256; ++i) {
+    const worker = new Worker(
+      'require(\'worker_threads\').parentPort.postMessage(2 + 2)',
+      { eval: true });
+    worker.on('message', (result) => {
+      assert.strictEqual(result, 4);
+    });
+    worker.on('error', (e) => {
+      assert(e.message.match(/Worker initialization failure: EMFILE/));
```

Comment:
```suggestion
      assert.match(e.message, /Worker initialization failure: EMFILE/);
```

Reply:
This assertion fails for me but the test still passes. Errors in the child are ignored. So this test isn't working, I'm afraid.

```console
$ ./node test/parallel/test-worker-init-failure.js 
assert.js:387
    throw err;
    ^

AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

  assert(e.message.match(/Worker initialization failure: EMFILE/))

    at Worker.<anonymous> (/Users/trott/io.js/test/parallel/test-worker-init-failure.js:19:7)
    at Worker.emit (events.js:321:20)
    at Worker.[kOnErrorMessage] (internal/worker.js:209:10)
    at Worker.[kOnMessage] (internal/worker.js:219:37)
    at MessagePort.<anonymous> (internal/worker.js:145:57)
    at MessagePort.emit (events.js:321:20)
    at MessagePort.onmessage (internal/worker/io.js:78:8)
    at MessagePort.exports.emitMessage (internal/per_context/messageport.js:11:10) {
  generatedMessage: true,
  code: 'ERR_ASSERTION',
  actual: null,
  expected: true,
  operator: '=='
}

$ tools/test.py test/parallel/test-worker-init-failure.js 
[00:00|% 100|+   1|-   0]: Done                            
$
```

Reply:
@Trott, I am unable to recreate this error. Did you happen to reduce file descriptors ?

Reply:
Change one of the assertions in the child process so that it fails no matter what and you should see this problem (unless I've done something horribly wrong here somehow).

Reply:
Looks like you added a check for the return code from the child process so I guess this particular problem is resolved.